app/Middleware/Queue/SubmissionService.php

```diff
@@ -167,10 +167,36 @@ public function getConsumerName()
     public function doSubmit(Message $message)
     {
         try {
-            $fh = fopen($message->file, 'r');
             $config = Config::getInstance();
             $config->set('CDASH_REMOTE_ADDR', $message->ip);
-            do_submit($fh, $message->project, null, $message->md5, $message->checksum);
+
+            $local_execution = file_exists($message->file);
+            if ($local_execution) {
+                // Local execution, process an open file.
+                $fh = fopen($message->file, 'r');
+            } else {
+                // Remote execution, pass the filename and CDash will retrieve
+                // its contents.
+                $fh = basename($message->file);
+            }
+            $handler = do_submit($fh, $message->project, null, $message->md5, $message->checksum);
+
+            if (!$local_execution) {
+                // If remotely executing, tell the server to delete the file
```

Comment:
Consider moving this logic to it's own method, e.g. `$service->delete($message)`. Ideally, to completely uncouple message deletion logic from the `SubmissionService::doSubmit`, it would be better to call `SubmissionService::delete` immediately after the `SubmissionService::doSubmit`, which probably requires a refactor resembling, e.g.:
```php
if ($service->doSubmit($message)) {
  $service->delete($message);
}
```
from the `SubmissionService::__call` method.